components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -61,4 +75,27 @@ export class GitLabAppSupport {
         result.unshift(...ownersRepos);
         return result;
     }
+
+    protected async getAccountAvatarUrl(project: ProjectSchema, providerHost: string): Promise<string> {
+        const owner = project.owner || project.namespace;
+        if (owner.avatar_url) {
+            const url = owner.avatar_url;
+            // Sometimes GitLab avatar URLs are relative -- ensure we always use the correct host
+            return url[0] === "/" ? `https://${providerHost}${url}` : url;
+        }
+        // If there is no avatar, generate the same default avatar that GitLab uses. Based on:
+        // - https://gitlab.com/gitlab-org/gitlab/-/blob/b2a22b6e85200ce55ab09b5c765043441b086c96/app/helpers/avatars_helper.rb#L151-161
+        // - https://gitlab.com/gitlab-org/gitlab-foss/-/blob/84b4743475246e91dc78c3f25f9b335c40be84cd/app/assets/stylesheets/startup/startup-general.scss#L1611-1631
+        // - https://gitlab.com/gitlab-org/gitlab-foss/-/blob/84b4743475246e91dc78c3f25f9b335c40be84cd/app/assets/stylesheets/startup/startup-general.scss#L420-422
+        const text = owner.name[0].toUpperCase();
+        const backgroundColor = ["#fcf1ef", "#f4f0ff", "#f1f1ff", "#e9f3fc", "#ecf4ee", "#fdf1dd", "#f0f0f0"][
+            owner.id % 7
```

Comment:
**issue**: One minor issue here is that some avatars are using a different background color from what GitLab does for each group. I cound not reproduce this for new groups but some of the existing groups. Is this expected?

| on GitLab.com | on Gitpod.io |
|-|-|
| <img width="384" alt="Screenshot 2022-03-28 at 10 12 54 PM" src="https://user-images.githubusercontent.com/120486/160470100-316716e6-df23-4eff-be64-3f798d35d27b.png"> | <img width="637" alt="Screenshot 2022-03-28 at 10 13 06 PM" src="https://user-images.githubusercontent.com/120486/160470101-8ef38242-b1b4-4270-b662-6ac59301d628.png"> |

Reply:
Interesting! Would have to look at debug logs -- maybe we're not using the right ID for these groups 👀

Reply:
Yeah, probably also minor issue that's not worth blocking us from merging this. 💭

Reply:
I think this might also be a case of "using the avatar of a nested group": https://github.com/gitpod-io/gitpod/pull/8824#discussion_r839377129

Reply:
This seems to be also resolved now!